package-res/resources/web/prompting/api/EventAPI.js

```diff
@@ -120,5 +120,21 @@ define([], function() {
         api.operation._getPromptPanel().ready = function(){};
       }
     };
+
+    /**
+     * Registers a listener for when the state of the prompt panel or parameter definition changes.
+     *
+     * @name EventAPI#stateChanged
+     * @method
+     * @param {Function} callback   The function to be executed when the event is triggered.
+     *                              Pass null if you wish to unbind this event.
+     * @example
+     *  api.event.stateChanged(function(name, oldValue, newValue) {
+     *    // Execute event based code
+     *  });
+     */
+    this.stateChanged = function(callback) {
+      api.operation._getPromptPanel().onStateChanged = callback;
```

Comment:
Not that i don't agree, but we are not doing it in any of the other calls, so we should do it for all.